data-host/library/src/lib/background.ts

```diff
@@ -1,5 +1,36 @@
+import {
+  getDataHost,
+  getMintingRegistrar,
+  getProtocolOptIn,
+} from '@services/Factory';
+
 export class Background {
   async setup(): Promise<void> {
     console.log('Background initialized.');
+    //TODO(melatron): Initialize everything needed for facts storage.
+  }
+  async addWebpage(url: string): Promise<void> {
+    await getProtocolOptIn().checkOptIn();
+    await getDataHost().storeFact({
+      pageView: {
+        url,
+        timestampMs: new Date().getTime(),
+      },
+    });
+
+    await getMintingRegistrar().maybeTryRegister();
+  }
+  addListeners() {
+    //TODO(melatron): Create more generic method that handles all type of messages (for different facts)
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this;
+    chrome.runtime.onMessage.addListener(function (
+      request,
+      _sender,
+      sendResponse
+    ) {
```

Comment:
Why is this a function and not a lambda? If you `addListener((request, ...) => {` you wouldn't need to `const self = this;`.

Reply:
`const self = this` is needed because the lambda is called in a different context. I will try if changing it from function to lambda would fix it.